### ISSUE TITLE
Failure verifying postconditions on bodiless payable functions.

### ISSUE THREAD

**Issue by michael-emmi** [SRI-CSL/solidity#111]
While bodiless non-payable functions like `f` below are happily translated to bodiless Boogie procedures, and verification skipped, bodiless payable functions like `g` are translated to Boogie procedures with bodies, thus triggering verification checks.
```
$ cat a.sol && echo --- && solc-verify.py a.sol --output . ; echo --- && cat a.sol.bpl 
pragma solidity ^0.5.0;

contract A {
    /// @notice postcondition r == 0
    function f() public returns (int r);

    /// @notice postcondition r == 0
    function g() public payable returns (int r);
}
---
A::g: ERROR
 - a.sol:8:5: Postcondition 'r == 0' might not hold at end of function.
A::[implicit_constructor]: OK
Errors were found by the verifier.
---
// Global declarations and definitions
type address_t = int;
var __balance: [address_t]int;
type string_t;
var __now: int;
var __block_number: int;
// 
// ------- Source: a.sol -------
// Pragma: solidity^0.5.0
// 
// ------- Contract: A -------
// 
// Function: f : function () returns (int256)
procedure {:sourceloc "a.sol", 5, 5} {:message "A::f"} f#6(__this: address_t, __msg_sender: address_t, __msg_value: int)
        returns (r#4: int);
        ensures {:sourceloc "a.sol", 5, 5} {:message "Postcondition 'r == 0' might not hold at end of function."} (r#4 == 0);


// 
// Function: g : function () returns (int256)
procedure {:sourceloc "a.sol", 8, 5} {:message "A::g"} g#11(__this: address_t, __msg_sender: address_t, __msg_value: int)
        returns (r#9: int)
        ensures {:sourceloc "a.sol", 8, 5} {:message "Postcondition 'r == 0' might not hold at end of function."} (r#9 == 0);

{
        // Update balance received by msg.value
        __balance := __balance[__this := (__balance[__this] + __msg_value)];
}

// 
// Default constructor
procedure {:sourceloc "a.sol", 3, 1} {:message "A::[implicit_constructor]"} __constructor#12(__this: address_t, __msg_sender: address_t, __msg_value: int)
{
        assume (__balance[__this] >= 0);
}
```

Ideally all bodiless functions (whether payable or not) would be translated to bodiless Boogie procedures, so that their checks are skipped.

**Comment by hajduakos** [comment]
Fixed in 2e44716b2de83f42dff2597855e02943236424a8

**Comment by michael-emmi** [comment]
Works now, thanks!
